# Assignment 6

To measure the psychological state of Americans, a psychology researcher (in 1929) devised
three questions put to 3443 Americans from different walks of life. He asked them whether they
were happy. They could answer only **yes** or **no** to each question. By today’s standards the study
is quite simple, and he did no formal statistical analysis of his data. The frequency data (number
of **yes** and **no** responses within each employment category) from this study are in the file
*Americandata.csv*.

---

You need to use the data and answer the question, that is, **Is the type of profession related to the happiness of American people?**
Report the **χ2** statistic and the *statistical significance of test*, in addition to the contingency table.
Also, explain the results and what they mean. What does the value of **χ2** statistic mean? Is it
good or bad? Is the test statistically significant?

In [6]:
#code here
import pandas as pd
import scipy.stats as stats

df = pd.read_csv("./Americandata.csv")

contingency_table = df.set_index('Happy')

chi2, p_value, dof, expected = stats.chi2_contingency(contingency_table, correction=False)

expected_df = pd.DataFrame(expected, index = contingency_table.index, columns = contingency_table.columns)

if (expected >=5).all():
  print("Assumption is okay.")
else:
  print("Assumption is not okay. Frequency counts in some cells are less than 5.")

print(f"\nChi-square = {chi2:.2f}")
print(f"\nDegrees of freedom = {dof}")
print(f"\np-value = {p_value:.2e}")

Assumption is okay.

Chi-square = 936.14

Degrees of freedom = 7

p-value = 7.52e-198


Put your analysis in the empty `markdown` cell below.

It is appropriate to use the chi-square test for the data since the code printed "Assumption is okay". The chi-square statistic, at 936.14, is very large which means it is way different than what would be expected if there was not relationship. It isn't necessarily good or bad. It is just a representation of the deviation from a "no relationship" hypthesis. The p-value is very small which just says that the test is statistically significant. Therefore, we reject the null-hypothesis as the results shown great evidence against the idea that profession and happiness are independent.